### PR TITLE
fix node graph unit tests

### PR DESCRIPTION
**Pull request by mm318**
Replaces GuidPrefix_t with DDS::InstanceHandle_t, because the meaning of the third word value in DDS::BuiltinTopicKey_t (and consequently GuidPrefix_t) is ambiguous.

### CODE REVIEW

File: rmw_opensplice_cpp/src/guid.hpp [hunk 1]
```diff
@@ -17,133 +17,19 @@
...
-  struct GuidPrefix_t * guid,
-  DDS::DomainId_t domain_id,
-  DDS::InstanceHandle_t instance_handle)
-{
-  memcpy(guid->value, reinterpret_cast<char *>(&domain_id), sizeof(DDS::DomainId_t));
-  memcpy(guid->value + sizeof(DDS::DomainId_t),
-    reinterpret_cast<char *>(&instance_handle), sizeof(DDS::InstanceHandle_t));
-}
 
 inline void DDS_BuiltinTopicKey_to_GUID(
```
Comment (ross-desmond) [inline rmw_opensplice_cpp/src/guid.hpp]: Change the function name as this is no longer what it does.
Rename the parameter

File: rmw_opensplice_cpp/src/guid.hpp [hunk 2]
```diff
@@ -17,133 +17,19 @@
...
-
-  bool operator!=(const GuidPrefix_t & prefix) const
-  {
-    return memcmp(value, prefix.value, kSize) != 0;
-  }
-
-#endif
-};
-
-inline bool operator<(const GuidPrefix_t & g1, const GuidPrefix_t & g2)
```
Comment (ross-desmond) [inline rmw_opensplice_cpp/src/guid.hpp]: Ensure that an instancehandle_t has these comparators. Required for stl containers.

File: rmw_opensplice_cpp/src/types.cpp [hunk 3]
```diff
@@ -186,8 +186,8 @@ print_discovery_logging(
 }
 
 void CustomDataReaderListener::add_information(
-  const GuidPrefix_t & participant_guid,
-  const GuidPrefix_t & topic_guid,
+  const DDS::InstanceHandle_t & participant_guid,
```
Comment (ross-desmond) [inline rmw_opensplice_cpp/src/types.cpp]: Note: if we decide to bloom a common package for discovery most of these would be a template or a typedef.

File: rmw_opensplice_cpp/src/guid.hpp (hunk 2 above)
Comment (mm318) [reply rmw_opensplice_cpp/src/guid.hpp]: InstanceHandle_t is a "long long", so it will have these comparators by default.

**Comment by mm318** [comment]
Justification of this change:

The `InstanceHandle_t` type does uniquely identify entities, such as `DomainParticipant`s, `Subscriber`s, `Publisher`s, and `Topic`s: https://github.com/ADLINK-IST/opensplice/blob/master/src/api/dcps/isocpp2/include/dds/core/TEntity.hpp#L203

So then the question is how to translate from `BuiltinTopicKey_t` to `InstanceHandle_t`.

We can first see that `BuiltinTopicKey`/`BuiltinTopicKey_t` can be translated to `v_builtinTopicKey`/`v_gid`: https://github.com/ADLINK-IST/opensplice/blob/master/src/api/dcps/isocpp2/code/org/opensplice/topic/BuiltinTopicCopy.cpp#L89

Then `v_builtinTopicKey`/`v_gid` can be translated to an `InstanceHandle_t` as done here: https://github.com/ADLINK-IST/opensplice/blob/master/src/api/dcps/c%2B%2B/common/code/DataReader.cpp#L1384 and here: https://github.com/ADLINK-IST/opensplice/blob/master/src/api/dcps/c%2B%2B/common/code/DataWriter.cpp#L552

This is the approach taken by the implementation of `DDS_BuiltinTopicKey_to_InstanceHandle()` in this pull request.

**Comment by ross-desmond** [comment]
Looks good to me

**Pull request closed without merging**